### PR TITLE
Add experimental Grok web reviewer

### LINKED ISSUE #70: Add experimental subscription-backed Grok web reviewer via local tunnel
## Context

We want to use an existing Grok / SuperGrok subscription for programmatic reviewer workflows without paying separately for xAI API usage. Official xAI API access appears to be separate from Grok subscription billing, so a direct `api.x.ai` provider is not the right fit for this use case.

There are unofficial subscription/session-backed projects that call Grok web endpoints using browser session cookies. After live testing, `chenyme/grok2api` is the preferred local tunnel target because it exposes OpenAI-compatible endpoints at `http://127.0.0.1:8000/v1`.

This is feasible, but it is not an official supported API. It should be isolated from normal direct API reviewers and treated as experimental.

## Current Direction

Add an experimental `grok-web` reviewer path that uses a local subscription-backed Grok tunnel, with `grok2api` as the documented/default tunnel shape.

Default transport shape:

- `GROK_WEB_BASE_URL=http://127.0.0.1:8000/v1`
- readiness probe: `GET /v1/models`
- review call: `POST /v1/chat/completions`
- request must set `stream: false` because this plugin expects a single JSON response object, not SSE
- default model should be basic-safe (`grok-4.20-fast`) unless the user selects a Super-tier model via `GROK_WEB_MODEL`

SuperGrok notes from live smoke:

- `grok2api` may auto-classify an imported SuperGrok session as `basic` even after quota refresh.
- Manually setting the local `grok2api` account pool to `super` made `grok-4.20-auto` and `grok-4.3-beta` work.
- The plugin should document this and the automation helper should support forcing the local pool to `super`.

## Proposed Commands / UX

Core reviewer commands:

- `/grok-setup`: local tunnel readiness check, reports key names only.
- `/grok-review`: read-only review over the selected scope.
- `/grok-adversarial-review`: adversarial review over the selected scope.
- `/grok-custom-review`: custom review over explicit scope/prompt.

Local session automation helper:

- Add a loud local helper such as `npm run grok:sync-browser-session`.
- Browser-cookie extraction should be the default setup path.
- The helper must explicitly announce which local browser/profile/cookie DB it is reading.
- It should target macOS Chrome-family browsers first and may require Keychain access.
- It should extract `sso-rw` preferentially and fall back to `sso`.
- It should import/replace the account in local `grok2api`, set pool to `super` by default, refresh quota, and print only sanitized status.
- It should verify local `grok2api` is reachable before touching browser cookies.
- It may offer clipboard fallback when browser extraction fails.

## Security Requirements

- Do not store raw Grok cookies in this repo's job records.
- Do not print raw cookies/session tokens in command output, logs, issue comments, test output, or JobRecords.
- Do not paste cookies into chat or require chat-mediated credential handoff.
- Redact or avoid secret-bearing values in persisted state.
- Direct `GROK_API_KEY` / xAI API keys are intentionally ignored for `subscription_web` and must not trigger paid API fallback.
- The helper may read local browser cookie storage only after loudly announcing what it will inspect.

## Non-goals

- Do not present this as official xAI API access.
- Do not silently fall back from subscription-backed Grok to paid xAI API billing.
- Do not implement broad, silent scraping across every browser/profile.
- Do not make browser-cookie automation cross-platform in the first pass; macOS Chrome-family support is enough.

## Risks

- Grok web endpoints may change without notice.
- Cookies expire or may trigger login/challenge/Cloudflare behavior.
- Browser cookie storage and Keychain access vary by browser/profile.
- Session cookies are sensitive and must not be logged or exposed.
- `grok2api` tier auto-detection may misclassify SuperGrok sessions.
- This may be outside the officially supported usage model, so docs must label it experimental and user-managed.

## Acceptance Criteria

- A user can configure or start a local `grok2api` tunnel and run a Grok web review through it.
- The resulting output is rendered through the same result UX used by other reviewers.
- Failure modes distinguish local tunnel unavailable, auth/session expired, rate/usage limits, malformed responses, and ignored direct xAI API keys where possible.
- Docs clearly explain that this uses subscription-backed web-session automation, not xAI API billing.
- Tests cover config validation, result mapping, `stream: false`, compatible model defaults, and secret redaction for cookie-like values.
- Live smoke passes against `grok2api` with a basic-safe model (`grok-4.20-fast`).
- SuperGrok smoke passes when the account pool is classified or forced as `super`, including `grok-4.20-auto` and `grok-4.3-beta`.
- Browser-session sync helper imports a local browser session without printing secrets and reports only sanitized account pool/quota status.

### PR DESCRIPTION
**Pull request by seungpyoson**
## Summary

Closes #70.

Adds an experimental `grok` plugin that delegates review workflows to a subscription-backed Grok web session through a local OpenAI-compatible tunnel. The default tunnel target is `grok2api` at `http://127.0.0.1:8000/v1`.

Key pieces:

- Adds Grok setup/review/adversarial/custom command docs and `grok:grok-delegation` skill.
- Adds `plugins/grok/scripts/grok-web-reviewer.mjs` with `doctor`, `ping`, and foreground review execution.
- Uses `subscription_web` auth mode and never falls back to paid `api.x.ai` billing.
- Defaults to `grok-4.20-fast` for basic-safe compatibility; higher-tier models can be selected with `GROK_WEB_MODEL`.
- Sends `stream: false` to `grok2api` so the plugin receives one JSON response object instead of SSE.
- Adds `npm run grok:sync-browser-session` for loud local macOS Chrome-family session sync into `grok2api`.
- Sync helper prefers `sso-rw`, falls back to `sso`, defaults local pool to `super`, refreshes quota, and prints only sanitized account status.
- Adds Grok smoke and live E2E coverage plus docs/runbook updates.

## Security Notes

- Raw session cookies are not printed or persisted in JobRecords.
- Direct `GROK_API_KEY` / xAI API keys are intentionally ignored for this subscription-backed path.
- The browser sync helper checks local `grok2api` reachability before reading local cookies and announces which browser profile/cookie DB it reads.

## Verification

- `npm run lint`
- `npm run smoke:grok`
- `node --test --test-reporter=spec tests/unit/docs-contracts.test.mjs tests/unit/ci-workflow.test.mjs tests/unit/manifests.test.mjs`
- `GROK_WEB_MODEL=grok-4.3-beta GROK_LIVE_E2E=1 npm run e2e:grok`
- `npm test`
- Pre-commit reran `npm test` successfully.

Live notes:

- Verified `npm run grok:sync-browser-session -- --browser chrome --profile Default --pool super` against local `grok2api`.
- The helper selected `sso-rw`, imported the session, forced `super`, refreshed quota, and printed sanitized quota only.

### CODE REVIEW

**Comment by seungpyoson** [comment]
## External review disposition for `da04aae`

Review range used for the second pass:

- Base: `395e09146fc9207afda80c11560d4e9f7709bf2d`
- Head: `da04aaed618608726aa5932b885c57dd7337f48e`

### Confirmed fixed

- **Token sync ordering / pool wipe risk**: fixed by changing grok2api sync to add + refresh the new token before deleting old tokens. Covered by `sync-browser-session does not delete existing tokens when add fails`.
- **grok2api admin API hangs**: fixed with per-call `AbortController` timeout and `GROK2API_ADMIN_TIMEOUT_MS` / `--admin-timeout-ms`. Covered by stalled admin-call smoke test.
- **Default grok2api admin key visibility**: fixed with stderr warning and runbook docs for `GROK2API_ADMIN_KEY`.
- **Git env injection in branch-diff**: fixed with `cleanGitEnv()` before Git invocations. Covered by branch-diff smoke test with hostile `GIT_DIR` / `GIT_WORK_TREE`.
- **`XAI_API_KEY` / `XAI_KEY` ignored-key diagnostic**: fixed alongside `GROK_API_KEY` diagnostic. Covered by doctor smoke test.
- **HTTP failure mappings**: covered for 401, 403, 429, 500, and malformed tunnel response.
- **Redaction gaps**: hardened for cookie fragments, bearer tokens, and authorization headers in both Grok reviewer and sync helper.
- **Docs gaps**: added Grok Web artifact-cleanup inventory and changelog entry.

### Reviewer-specific notes

- Claude, Kimi, and DeepSeek re-review: no blocking issues remain.
- Gemini’s remaining “workflow skills” finding appears to target the workflow-skills convention from issue/PR #71, not PR #72’s current base. On this PR’s actual base, sibling providers still expose broad delegation skills rather than per-workflow skills. I am treating this as dependent on #71: if #71 lands first, PR #72 should rebase and add Grok per-workflow skills for parity.

### Remaining work before final landing

The team wants no residual risks before merge, so these are not being accepted as final residuals:

- Add or explicitly settle Chrome cookie decrypt/keychain regression coverage.
- Add a file-size cap for selected scope reads to avoid accidental quota burn.
- Add a plugin-side way to read Grok JobRecords, or remove/purpose-limit the persistence if status/result parity is not intended.
- Tighten any remaining redaction/env-passthrough edge cases found during the final cleanup pass.

Current verification at `da04aae`:

- Local `npm test`: `816` tests, `810` pass, `6` skipped.
- Local `npm run lint`: pass.
- Local `git diff --check`: pass.
- GitHub checks: SonarCloud, lint, smoke matrix, and full test all pass.

File: plugins/grok/scripts/grok-web-reviewer.mjs [hunk 1]
```diff
@@ -0,0 +1,593 @@
...
+    }
+    const abs = resolve(workspaceRoot, relPath);
+    const normalizedRel = relative(workspaceRoot, abs);
+    if (normalizedRel.startsWith("..") || normalizedRel === "") {
+      throw new Error(`unsafe_scope_path:${relPath}`);
+    }
+    if (!existsSync(abs)) continue;
+    const text = await readFile(abs, "utf8");
+    if (text.length === 0) continue;
+    files.push({ path: normalizedRel, text });
```
[the remark this answers is not in the document]
Comment (seungpyoson) [reply plugins/grok/scripts/grok-web-reviewer.mjs]: Addressed in `c3c3a7a`.

Current head enforces `MAX_SCOPE_FILE_BYTES = 256 * 1024` before reading selected files, so oversized scope files fail with `scope_failed` and `source_content_transmission: "not_sent"` before the tunnel is contacted. Covered by `custom-review rejects oversized selected files before contacting the tunnel`.

File: plugins/grok/scripts/grok-sync-browser-session.mjs [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+  }
+
+  const selected = selectCookie(cookies);
+  if (!selected) {
+    fail("cookie_not_found", "No usable sso-rw or sso cookie was found for grok.com.", { source });
+  }
+
+  const toDelete = existingTokens.map((entry) => entry.token).filter(Boolean);
+  try {
+    if (toDelete.length) {
```
[the remark this answers is not in the document]
Comment (seungpyoson) [reply plugins/grok/scripts/grok-sync-browser-session.mjs:350]: Addressed in `c3c3a7a`.

The sync helper no longer deletes every existing grok2api token. Replacement is scoped to the selected target pool and excludes the newly synced token; accounts in other pools are preserved. Added `--append` for users who want to keep existing accounts in the same target pool as well. Covered by `sync-browser-session imports sso-rw into grok2api, forces super pool, and redacts token output` and `sync-browser-session append mode keeps existing same-pool tokens`.

File: plugins/grok/scripts/grok-web-reviewer.mjs [hunk 3]
```diff
@@ -0,0 +1,593 @@
...
+function promptFor(mode, userPrompt, scopeInfo) {
+  const modeLine = mode === "adversarial-review"
+    ? "You are performing an adversarial code review. Prioritize correctness bugs, security risks, regressions, and missing tests."
+    : "You are performing a code review. Prioritize bugs, behavioral regressions, and missing tests.";
+  const files = scopeInfo.files.map((file) => [
+    `### ${file.path}`,
+    "```",
+    file.text,
+    "```",
+  ].join("\n")).join("\n\n");
```
[the remark this answers is not in the document]
Comment (seungpyoson) [reply plugins/grok/scripts/grok-web-reviewer.mjs]: Addressed in `c3c3a7a`.

Prompt file blocks no longer use triple-backtick fences. They now use per-file `BEGIN GROK FILE ...` / `END GROK FILE ...` delimiters, with delimiter adjustment if the file content already contains the marker. The smoke test now includes nested triple backticks in selected source and asserts the outbound prompt has no standalone backtick fence.

**Comment by seungpyoson** [comment]
Greptile current summary item addressed in `c3c3a7a`.

- `grok-web-reviewer.mjs list` now returns a bounded recent-job index instead of only the latest run. `persistRecord` reads existing `state.json`, prepends the current job summary, deduplicates by job id, and keeps the newest 50 entries.
- Docs now describe `state.json` as a recent-job index, not a latest-job shortcut.
- Added smoke coverage by running two Grok custom-review invocations against the same `GROK_PLUGIN_DATA` and asserting `list` returns both jobs newest-first.

I also rechecked the previously referenced Greptile risks while addressing this pass: selected files still have a 256 KiB pre-send cap, and this commit adds a regression test for symlinks that resolve outside the workspace so those are rejected before any tunnel call.

File: plugins/grok/scripts/grok-web-reviewer.mjs [hunk 4]
```diff
@@ -0,0 +1,939 @@
...
+    const info = await stat(abs);
+    if (!info.isFile()) continue;
+    if (info.size > MAX_SCOPE_FILE_BYTES) {
+      throw new Error(`scope_file_too_large:${normalizedRel}: ${info.size} bytes exceeds ${MAX_SCOPE_FILE_BYTES} byte limit`);
+    }
+    totalBytes += info.size;
+    if (totalBytes > MAX_SCOPE_TOTAL_BYTES) {
+      throw new Error(`scope_total_too_large:${totalBytes} bytes exceeds ${MAX_SCOPE_TOTAL_BYTES} byte limit`);
+    }
+    const text = await readFile(abs, "utf8");
```
[the remark this answers is not in the document]
Comment (seungpyoson) [reply plugins/grok/scripts/grok-web-reviewer.mjs]: Fixed in `7e9f9d27c46e8fd4a0bc0ffb86516894bf1b2479`.

`readScopeFiles` now uses the canonical `realAbs` for both `stat` and `readFile` after the realpath workspace-boundary check, so the verified path is the path that gets inspected and read.

I also fixed two related Greptile summary items in the same commit:
- `branch-diff` now uses `${base}...HEAD`, and the smoke test proves a dirty tracked base-branch file containing `GROK_LOCAL_DIRTY_SECRET` is not selected or sent to the tunnel.
- prompt construction failures remain `scope_failed`, while unexpected tunnel invocation throws are now separated as `tunnel_error`.

File: plugins/grok/scripts/grok-sync-browser-session.mjs [hunk 5]
```diff
@@ -0,0 +1,383 @@
...
+  const encrypted = Buffer.from(encryptedHex, "hex");
+  if (!encrypted.length) return "";
+  if (!(encrypted.subarray(0, 3).toString("utf8") === "v10" || encrypted.subarray(0, 3).toString("utf8") === "v11")) {
+    throw new Error("unsupported encrypted cookie format");
+  }
+  const payload = encrypted.subarray(3);
+  const key = pbkdf2Sync(Buffer.from(password, "utf8"), Buffer.from("saltysalt"), 1003, 16, "sha1");
+  const iv = Buffer.alloc(16, " ");
+  const decipher = createDecipheriv("aes-128-cbc", key, iv);
+  return Buffer.concat([decipher.update(payload), decipher.final()]).toString("utf8");
```
[the remark this answers is not in the document]
Comment (seungpyoson) [reply plugins/grok/scripts/grok-sync-browser-session.mjs:272]: Disproved for this PR's supported path; no code change needed.

This helper is macOS-specific: it reads the macOS Chrome cookie DB and obtains Chrome Safe Storage through `/usr/bin/security`/Keychain. Chromium's macOS OSCrypt implementation still identifies the Keychain-backed ciphertext prefix as `v10`: https://chromium.googlesource.com/chromium/src/+/0267e3c2/components/os_crypt/os_crypt_mac.mm

The Chrome 127 App-Bound Encryption migration Greptile is pointing at is explicitly a Windows protection: https://security.googleblog.com/2024/07/improving-security-of-chrome-cookies-on.html

So a Windows `v20` decryptor does not belong in this macOS session-sync helper. For unsupported/exported-cookie workflows the command already has `--cookie-source-json`, which avoids local Chrome cookie decryption entirely.

**Comment by seungpyoson** [comment]
Greptile/Gemini review comment follow-up at `7e9f9d27c46e8fd4a0bc0ffb86516894bf1b2479`:

- Greptile TOCTOU on `readScopeFiles`: fixed by using verified `realAbs` for both `stat` and `readFile`; inline thread replied and resolved.
- Greptile branch-diff scope summary item: fixed by switching `branch-diff` selection to `${base}...HEAD`; smoke coverage now proves dirty tracked base-branch files are not sent to the tunnel.
- Greptile tunnel-error classification summary item: fixed by separating prompt construction errors (`scope_failed`) from unexpected tunnel invocation throws (`tunnel_error`).
- Greptile Chrome `v20` cookie thread: disproved as not applicable to this macOS Keychain helper; replied with Chromium macOS OSCrypt and Chrome Windows App-Bound Encryption source context, then resolved.
- Gemini Code Assist: no unresolved threads remained when checked.

Verification for the exact committed patch was run in a clean temporary worktree:
- `node --test tests/smoke/grok-web.smoke.test.mjs` -> 43/43 pass
- `node --test tests/smoke/grok-session-sync.smoke.test.mjs tests/unit/manifests.test.mjs tests/unit/plugin-copies-in-sync.test.mjs tests/unit/docs-contracts.test.mjs` -> 73/73 pass
- `npm test` -> 861 total, 855 pass, 6 skipped, 0 fail

Note: the main local worktree has unrelated unstaged model/review-prompt changes; a full `npm test` there fails on those unrelated changes. The clean-worktree run above verifies the PR commit itself.

**Comment by seungpyoson** [comment]
Post-merge follow-up: PR #72 was merged before I could push the broader review-quality fixes for issues #73/#74. I rebased those changes onto current `main` and opened #75.

Relevant PR #72 review-comment handling:
- the symlink TOCTOU path is covered on merged `main` by reading/statting through the canonical real path after the boundary check;
- the Grok tunnel/prompt error classification and branch-diff scope cleanup remain in the merged Grok path;
- Chrome `v20` cookie handling is now addressed in #75 with an explicit actionable unsupported-format failure plus tests, rather than attempting unsafe/incorrect Safe Storage decryption.
